src/PageContent/AssignVolunteers/AssignVolunteersController.js

```diff
@@ -59,17 +60,22 @@ class AssignVolunteersController extends Component {
         let processTimestampIndex = (timestampIndex) => {
             for (let deliveryId of Object.keys(timestampIndex)) {
                 if (!this.state.deliveries[deliveryId]) {
-                    if (!this.state.deliveriesExist) {
-                        this.setState({deliveriesExist: true});
-                    }
                     genDeliveryListener(deliveryId);
                 }
             }
         };
+
+        let updateLoading = (children, myDeliveriesRef) => {
+            if(children) { this.setState({deliveriesExist : true}); }
+            this.setState({finishedCall : true});
+        };
+
         myDeliveriesRef.on('child_added', async (snap) => processTimestampIndex(snap.val()));
         myDeliveriesRef.on('child_changed', async (snap) => processTimestampIndex(snap.val()));
+        myDeliveriesRef.once('value', (snap) => updateLoading(snap.val(), myDeliveriesRef));
```

Comment:
I think we should process existing deliveries here, since the timestamp and delivery data comes down in the snap anyway. Maybe add something like 
```
if(children) { 
  this.setState({deliveriesExist : true});
  for (let timestamp of Object.keys(children)) {
    processTimestampIndex(children[timestamp]);
  }
}
```

in `update_loading`. This in conjunction with the suggestion for line 73 should improve the efficiency of this query.